client/landing/jetpack-cloud/components/sidebar/index.jsx

```diff
@@ -0,0 +1,202 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Sidebar from 'layout/sidebar';
+import SidebarFooter from 'layout/sidebar/footer';
+import SidebarItem from 'layout/sidebar/item';
+import SidebarMenu from 'layout/sidebar/menu';
+import SidebarRegion from 'layout/sidebar/region';
+import ExpandableSidebarMenu from 'layout/sidebar/expandable';
+
+class JetpackCloudSidebar extends Component {
+	static propTypes = {
+		context: PropTypes.shape( {
+			path: PropTypes.string,
+		} ),
+	};
+
+	state = {
+		expandedSections: [],
+	};
+
+	componentDidMount() {
+		const section = this.getSectionFromCurrentPath();
+
+		if ( section ) {
+			this.toggleSection( section );
+		}
+	}
+
+	/**
+	 * Get current section name from path. When there's no section, we're in the dashboard.
+	 *
+	 * @returns {string} Current section
+	 */
+	getSectionFromCurrentPath() {
+		// If no section is present in the path, return 'dashboard'
+		return this.props.context.path.split( '/' )[ 2 ] ?? 'dashboard';
+	}
+
+	/**
+	 * Get current subsection name from path.
+	 *
+	 * @returns {string|null} Current subsection or null
+	 */
+	getSubsectionFromCurrentPath() {
+		return this.props.context.path.split( '/' )[ 3 ] ?? null;
+	}
+
+	/**
+	 * Toggle an expandable section.
+	 *
+	 * @param   {string} section Section to be toggled
+	 * @returns {void}
+	 */
+	toggleSection( section ) {
+		if ( this.isExpanded( section ) ) {
+			this.setState( {
+				expandedSections: this.state.expandedSections.filter( item => item !== section ),
+			} );
+		} else {
+			this.setState( {
+				expandedSections: [ ...this.state.expandedSections, section ],
+			} );
+		}
+	}
+
+	/**
+	 * Check if a section menu is expanded.
+	 *
+	 * @param   {string} section Section name
+	 * @returns {boolean}        Is the section expanded
+	 */
+	isExpanded( section ) {
+		return this.state.expandedSections.includes( section );
+	}
+
+	/**
+	 * Check if a menu item is selected based on the current path.
+	 *
+	 * @param   {string}  path Menu item path
+	 * @returns {boolean}      Is the menu item selected
+	 */
+	isSelected( path ) {
+		const section = this.getSectionFromCurrentPath();
+		const subsection = this.getSubsectionFromCurrentPath();
+
+		if ( ! subsection ) {
+			return section === path;
+		}
+
+		return section.concat( '/', subsection ) === path;
+	}
+
+	handleExpandableMenuClick( section ) {
+		return () => this.toggleSection( section );
+	}
+
+	onNavigate = () => {
+		window.scrollTo( 0, 0 );
+	};
+
+	render() {
+		return (
+			<Sidebar>
+				<SidebarRegion>
+					{ /* @todo: A profile info box needs to be created and added here; similar to <ProfileGravatar /> in client/me/sidebar/index.jsx */ }
+					<SidebarMenu>
+						<ul>
```

Comment:
Are the nested `UL`  necessary?

Reply:
Thanks for catching that up! I copied that structure from [other sidebar implementation](https://github.com/Automattic/wp-calypso/blob/de18db59f34af488ff47092e28b596ad9dade68a/client/me/sidebar/index.jsx#L136) but I didn't really check out the resulting DOM. This can definitely be dropped.

Reply:
It turns out that this UL is needed in order to have at least some HTML semantics out there.

`SiebarItem` returns a LI while `ExpandableSidebarMenu` renders its `children` inside a DIV.  The `ExpandableSidebarMenu` itself is rendered as a UL, with a H2 and a DIV inside.

Here is a source of two menus: the regular one followed by an expandable menu:

<img width="395" alt="Screenshot 2020-01-22 at 17 31 24" src="https://user-images.githubusercontent.com/478735/72913187-11411a00-3d3d-11ea-80c2-a7a3f225dd1a.png">

I guess this kind of discrepancies should be fixed in a separate PR.